includes/Gateway/Cash_App_Pay_Gateway.php

```diff
@@ -0,0 +1,1226 @@
+				if ( $this->debug_log() ) {
+					$this->get_plugin()->log( $exception->getMessage(), $this->get_id() );
+				}
+			}
+		}
+
+		return $order;
+	}
+
+	/**
+	 * Gets an order with refund data attached.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int|\WC_Order $order order object
+	 * @param float $amount amount to refund
+	 * @param string $reason response for the refund
+	 *
+	 * @return \WC_Order|\WP_Error
+	 */
+	protected function get_order_for_refund( $order, $amount, $reason ) {
+
+		$order                 = parent::get_order_for_refund( $order, $amount, $reason );
+		$order->square_version = $this->get_order_meta( $order, 'square_version' );
+		$transaction_date      = $this->get_order_meta( $order, 'trans_date' );
+
+		if ( $transaction_date ) {
+			// refunds with the Refunds API can be made up to 1 year after payment and up to 120 days with the Transactions API
+			$max_refund_time = version_compare( $order->square_version, '2.2', '>=' ) ? '+1 year' : '+120 days';
+
+			// throw an error if the payment cannot be refunded
+			if ( time() >= strtotime( $max_refund_time, strtotime( $transaction_date ) ) ) {
+				/* translators: %s maximum refund date. */
+				return new \WP_Error( 'wc_square_refund_age_exceeded', sprintf( __( 'Refunds must be made within %s of the original payment date.', 'woocommerce-square' ), '+1 year' === $max_refund_time ? 'a year' : '120 days' ) );
+			}
+		}
+
+		$order->refund->location_id = $this->get_order_meta( $order, 'square_location_id' );
+		$order->refund->tender_id   = $this->get_order_meta( $order, 'authorization_code' );
+
+		if ( ! $order->refund->tender_id ) {
+
+			try {
+				$response = version_compare( $order->square_version, '2.2', '>=' ) ? $this->get_api()->get_payment( $order->refund->trans_id ) : $this->get_api()->get_transaction( $order->refund->trans_id, $order->refund->location_id );
+
+				if ( ! $response->get_authorization_code() ) {
+					throw new \Exception( 'Tender missing' );
+				}
+
+				$this->update_order_meta( $order, 'authorization_code', $response->get_authorization_code() );
+				$this->update_order_meta( $order, 'square_location_id', $response->get_location_id() );
+
+				$order->refund->location_id = $response->get_location_id();
+				$order->refund->tender_id   = $response->get_authorization_code();
+
+			} catch ( \Exception $exception ) {
+
+				return new \WP_Error( 'wc_square_refund_tender_missing', __( 'Could not find original transaction tender. Please refund this transaction from your Square dashboard.', 'woocommerce-square' ) );
+			}
+		}
+
+		return $order;
+	}
+
+	/**
+	 * Gets the configured environment ID.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return string
+	 */
+	public function get_environment() {
+		return self::ENVIRONMENT_PRODUCTION;
+	}
+
+	/**
+	 * Gets the configured application ID.
+	 *
+	 * @since x.x.x
+	 *
+	 * @return string
+	 */
+	public function get_application_id() {
+		$square_application_id = 'sq0idp-wGVapF8sNt9PLrdj5znuKA';
```

Comment:
Is this something that's truly standard across all merchants and should be stored in the code?

Reply:
Yes, it's the default application ID for all merchants, as we use connect.woocommerce.com to set up the OAuth connection for the Production environment. 

> should be stored in the code? 

I followed the same approach we currently have for the Square Credit Card gateway here. I don't see any concern here to have this hardcoded. @dkotter any thoughts?

Reply:
Yeah, at first glance seems potentially unsafe to have this but this isn't something newly introduced here, this is how the main Square Gateway functions. Looking into the history here, this was introduced ~5 years ago and there was a call out on this that was ultimately okayed: https://github.com/woocommerce/woocommerce-square-private/pull/265#discussion_r328320283